app/bundles/LeadBundle/Model/TagModel.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Model;
+
+use Mautic\CoreBundle\Model\FormModel;
+use Mautic\LeadBundle\Entity\Tag;
+use Mautic\LeadBundle\Event\TagEvent;
+use Mautic\LeadBundle\Form\Type\TagEntityType;
+use Mautic\LeadBundle\LeadEvents;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException;
+
+/**
+ * Class TagModel
+ * {@inheritdoc}
+ */
+class TagModel extends FormModel
```

Comment:
Is it necessary to extends this model? New code should be as clean as possible. Do you really need any functionality from it?

Reply:
Yep. Look at https://github.com/mautic/mautic/blob/staging/app/bundles/ApiBundle/Controller/CommonApiController.php and search for `model->`. I need all those methods and those are in the Form Model.